components/bl-avatar-generator/src/helpers.js

```diff
@@ -0,0 +1,223 @@
+import { data as characterOptions } from './avatar-data';
+
+const RANDOM_OPTION = 'Random';
+
+const OptionsKeys = {
+  BACKGROUND        : 'background',
+  CIRCLE_COLOR      : 'circleColor',
+  SKIN              : 'skinColors',
+  TOP               : 'topTypes',
+  HAIR_COLOR        : 'hairColors',
+  HAT_COLOR         : 'hatColors',
+  BROWS             : 'browsTypes',
+  EYES              : 'eyesTypes',
+  MOUTH             : 'mouthTypes',
+  FACIAL_HAIR       : 'facialHairTypes',
+  FACIAL_HAIR_COLOR : 'hairColors',
+  ACCESSORY         : 'accessoryTypes',
+  CLOTHES           : 'clothesTypes',
+  FABRIC_COLOR      : 'hatColors',
+  GRAPHIC           : 'graphicTypes',
+};
+
+// the order is important for handleRandomOptions
+const ComponentOptionsKeys = {
+  BACKGROUND        : 'background',
+  CIRCLE_COLOR      : 'circleColor',
+  SKIN              : 'skin',
+  TOP               : 'top',
+  HAIR_COLOR        : 'hairColor',
+  HAT_COLOR         : 'hatColor',
+  BROWS             : 'brows',
+  EYES              : 'eyes',
+  MOUTH             : 'mouth',
+  FACIAL_HAIR       : 'facialHair',
+  FACIAL_HAIR_COLOR : 'facialHairColor',
+  ACCESSORY         : 'accessory',
+  CLOTHES           : 'clothes',
+  FABRIC_COLOR      : 'fabricColor',
+  GRAPHIC           : 'graphic',
+};
+
+export const DefaultOptions = {
+  BACKGROUND       : 'transparent',
+  CIRCLE_COLOR     : '#E6E6E6',
+  SKIN             : getRandomValue(OptionsKeys.SKIN),
+  TOP              : getRandomValue(OptionsKeys.TOP),
+  HAIR_COLOR       : getRandomValue(OptionsKeys.HAIR_COLOR),
+  HAT_COLOR        : getRandomValue(OptionsKeys.HAT_COLOR),
+  BROWS            : getRandomValue(OptionsKeys.BROWS),
+  EYES             : getRandomValue(OptionsKeys.EYES),
+  MOUTH            : getRandomValue(OptionsKeys.MOUTH),
+  FACIAL_HAIR      : getRandomValue(OptionsKeys.FACIAL_HAIR),
+  FACIAL_HAIR_COLOR: getRandomValue(OptionsKeys.FACIAL_HAIR_COLOR),
+  ACCESSORY        : getRandomValue(OptionsKeys.ACCESSORY),
+  CLOTHES          : getRandomValue(OptionsKeys.CLOTHES),
+  FABRIC_COLOR     : getRandomValue(OptionsKeys.FABRIC_COLOR),
+  GRAPHIC          : getRandomValue(OptionsKeys.GRAPHIC),
```

Comment:
need a function that accept keys which need to generate and source object(OptionsKeys)